File: pms.backend/pms.backend/Models/Projects/ProjectsModel.cs
using System.ComponentModel.DataAnnotations;

namespace pms.backend.Models.Projects
{
    public class ProjectsModel : IProjectsModel
    {
        [Key]
        public int ProjectId { get; set; }
        public string ProjectName { get; set; }
        public string ProjectDescription { get; set; }
        public string ProjectOwnerId { get; set; }
        public DateTime ProjectDate { get; set; }
    }
}

File: pms.frontend/src/models/project/ProjectsModel.ts
interface IProjectModelShape {
    projectId: number,
    projectName: string,
    projectDescription: string,
    projectOwnerId: number,
    projectDate: Date
}

export default class ProjectModel implements IProjectModelShape{
    projectId;
    projectName;
    projectDescription;
    projectOwnerId;
    projectDate;
    constructor(opts: IProjectModelShape) {
        this.projectId = opts.projectId;
        this.projectName = opts.projectName;
        this.projectDescription = opts.projectDescription;
        this.projectOwnerId = opts.projectOwnerId;
        this.projectDate = opts.projectDate;
    }
}